pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstance.java

```diff
@@ -61,8 +67,21 @@ public JavaInstance(InstanceConfig config, Object userClassObject,
         // create the functions
         if (userClassObject instanceof Function) {
             this.function = (Function) userClassObject;
-        } else {
-            this.javaUtilFunction = (java.util.function.Function) userClassObject;
+        } else if (userClassObject instanceof java.util.function.Function) {
+            Class<?>[] typeArgs = TypeResolver.resolveRawArguments(
+                    java.util.function.Function.class, userClassObject.getClass());
+            // check if window function
+            if (typeArgs[0].equals(Collection.class)) {
```

Comment:
I believe this check should be different? We should check if windowConfig is set, then do this.
I could have a regular function that actually takes in a Collection

Reply:
yup good catch!

Reply:
though its a bit difficult to get the check if the windowconfig set since the windowconfig is client side and not on the server side